doc/reference/kernel/threads/workqueue.rst

```diff
@@ -147,6 +151,51 @@ may or may not be invoked.
    must check return values and be prepared to react when either submission or
    cancellation fails.
 
+.. _k_delayable_work:
+
+Delayable Work
```

Comment:
Sorry for being so late in the game and perhaps this has been discussed already in the PR. The `k_delayable_work` is almost identical to `k_delayed_work`, was there any consideration to name the new one as `k_scheduled_work` as that would look similar to new scheduled API naming.

Reply:
I think it started out that way, but as I came to understood the core concepts I changed it.  Thinking of the life cycle as "scheduled" then "submitted" is fundamental to understanding what can and cannot be done: i.e. it can be cancelled only if it's scheduled, not if it's submitted.

The structure may or may not be scheduled (it can go directly to pending) so I think it's accurate to name it as a work item that supports being delayed.

Also `k_work_schedule` pairs with `k_work_cancel` nicely.  Making it `k_scheduled_work_submit()` and `k_scheduled_work_cancel()` to follow the consistent-prefix rule is (a) longer which annoys some people, and (b) doesn't as clearly distinguish the two phases of scheduled and submitted (IMO) (and `k_scheduled_work_schedule()` seems redundant).